builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -565,32 +560,6 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 		}
 	}
 
-	if _, ok := d.GetOk("network_interface.0.ipv4_address"); !ok {
-		if v, ok := d.GetOk("boot_delay"); ok {
-			stateConf := &resource.StateChangeConf{
-				Pending:    []string{"pending"},
-				Target:     []string{"active"},
-				Refresh:    waitForNetworkingActive(client, vm.datacenter, vm.Path()),
-				Timeout:    600 * time.Second,
-				Delay:      time.Duration(v.(int)) * time.Second,
-				MinTimeout: 2 * time.Second,
-			}
-
-			_, err := stateConf.WaitForState()
-			if err != nil {
-				return err
-			}
-		}
-	}
-
-	if ip, ok := d.GetOk("network_interface.0.ipv4_address"); ok {
-		d.SetConnInfo(map[string]string{
-			"host": ip.(string),
-		})
```

Comment:
Already done in resourceVSphereVirtualMachineRead (called at the end). Both waiting for network and "SetConnInfo"